### PR TITLE
Add IO resource control settings for Windows

### PR DESCRIPTION
**Pull request by darrenstahlmsft**
Signed-off-by: Darren Stahl <darst@microsoft.com>

### CODE REVIEW

File: types/container/host_config.go [hunk 1]
```diff
@@ -181,13 +182,17 @@ type Resources struct {
...
 	KernelMemory         int64           // Kernel memory limit (in bytes)
-	Memory               int64           // Memory limit (in bytes)
 	MemoryReservation    int64           // Memory soft limit (in bytes)
 	MemorySwap           int64           // Total memory usage (memory + swap); set `-1` to enable unlimited swap
 	MemorySwappiness     *int64          // Tuning container memory swappiness behaviour
 	OomKillDisable       *bool           // Whether to disable OOM Killer or not
 	PidsLimit            int64           // Setting pids limit for a container
 	Ulimits              []*units.Ulimit // List of ulimits to be set in the container
+
+	// Applicable to Windows platforms
```
Comment (jhowardmsft) [inline types/container/host_config.go]: Just Windows :smile:

**Comment by jhowardmsft** [comment]
Minor nit, otherwise LGTM

File: types/container/host_config.go (hunk 1 above)
Comment (darrenstahlmsft) [inline types/container/host_config.go]: Fair

**Comment by calavera** [comment]
@jhowardmsft I wonder if we should have two different structures by platform. We've talked about this in the past and it might be a good moment to do this, since most of the fields in the `Resources` struct don't apply to both. Is it a good idea to have a `CommonResources` with shared fields and a `Resources` struct by platform?

**Comment by jhowardmsft** [comment]
@calavera Unfortunately we can't. I had a PR that did this a long time back, but realised it wasn't workable. The reason is that this is a common structure passed from client to daemon. The client has no idea whether the daemon is Windows or Linux, so has to pass all fields. Otherwise a Windows client wouldn't be able to correctly communicate with a Linux daemon, or visa-versa.

**Comment by calavera** [comment]
> The client has no idea whether the daemon is Windows or Linux, so has to pass all fields

:pensive: :penguin: 

LGTM

**Pull request merged**